### PR TITLE
Python: Exclude certificate classification fo sensitive data queries

### PR DESCRIPTION
**Pull request by joefarebrother**
A certificate, such as an SSL certificate or x509 certificate, often does not contain sensitive data, so the cleartext storage and cleartext logging queries result in false positive alerts when considering them. This PR excludes certificates as sources for these queries.

### CODE REVIEW

File: python/ql/src/change-notes/2024-08-27-sensitive-certificate.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* The `py/clear-text-logging-sensitive-data` and `py/clear-text-storage-sensitive-data` queries have been updated to exclude the `certificate` classification of of sensitive sources, which often do not actually contain sensitive data.
```
Comment (sidshank) [inline python/ql/src/change-notes/2024-08-27-sensitive-certificate.md:4]:
```suggestion
* The `py/clear-text-logging-sensitive-data` and `py/clear-text-storage-sensitive-data` queries have been updated to exclude the `certificate` classification of sensitive sources, which often do not contain sensitive data.
```

File: python/ql/test/query-tests/Security/CWE-312-CleartextStorage-py3/test.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
 import pathlib
 
 
-def get_cert():
-    return "<CERT>"
+def get_password():
+    return "password"
```
[the remark this answers is not in the document]
Comment (sidshank) [reply python/ql/test/query-tests/Security/CWE-312-CleartextStorage-py3/test.py:5]: I'll defer to @RasmusWL and @tausbn's judgement on that. 🙏🏼

File: python/ql/lib/semmle/python/security/dataflow/CleartextLoggingCustomizations.qll [hunk 3]
```diff
@@ -41,7 +41,8 @@ module CleartextLogging {
    */
   class SensitiveDataSourceAsSource extends Source, SensitiveDataSource {
     SensitiveDataSourceAsSource() {
-      not SensitiveDataSource.super.getClassification() = SensitiveDataClassification::id()
+      not SensitiveDataSource.super.getClassification() =
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/security/dataflow/CleartextLoggingCustomizations.qll:44]: NIT: we usually prefer `in [...]` for readability, although using `= [...]` is the same :shrug: 

```suggestion
      not SensitiveDataSource.super.getClassification() in
```

File: python/ql/lib/semmle/python/security/dataflow/CleartextStorageCustomizations.qll [hunk 4]
```diff
@@ -40,7 +40,8 @@ module CleartextStorage {
    */
   class SensitiveDataSourceAsSource extends Source, SensitiveDataSource {
     SensitiveDataSourceAsSource() {
-      not SensitiveDataSource.super.getClassification() = SensitiveDataClassification::id()
+      not SensitiveDataSource.super.getClassification() =
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/security/dataflow/CleartextStorageCustomizations.qll:43]:
```suggestion
      not SensitiveDataSource.super.getClassification() in
```

File: python/ql/test/query-tests/Security/CWE-312-CleartextStorage-py3/test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (RasmusWL) [reply python/ql/test/query-tests/Security/CWE-312-CleartextStorage-py3/test.py:5]: My interpretation is that this `-py3` test is here to ensure we properly support writing through `pathlib.Path` methods for this query, so LGTM :+1:

**Review by RasmusWL — COMMENTED** [review]
Besides minor style NIT, looks good to me :+1:

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**